packages/terminal-external/src/electron-node/terminal-external-service.ts

```diff
@@ -0,0 +1,195 @@
+/********************************************************************************
+ * Copyright (C) 2021 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as cp from 'child_process';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { injectable } from 'inversify';
+import { OS } from '@theia/core/lib/common/os';
+import { FileUri } from '@theia/core/lib/node/file-uri';
+import { TerminalExternalService, TerminalExternalConfiguration } from '../common/terminal-external';
+
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+// some code copied and modified from https://github.com/microsoft/vscode/blob/1.52.1/src/vs/workbench/contrib/externalTerminal/node/externalTerminalService.ts
+
+@injectable()
+export class WindowsTerminalExternalService implements TerminalExternalService {
+    private static readonly CMD = 'cmd.exe';
+
+    private static DEFAULT_TERMINAL_WINDOWS: string;
+
+    async openTerminal(configuration: TerminalExternalConfiguration, cwd: string): Promise<void> {
+        await this.spawnTerminal(configuration, FileUri.fsPath(cwd));
+    }
+
+    private async spawnTerminal(configuration: TerminalExternalConfiguration, cwd?: string): Promise<void> {
```

Comment:
I think it's good as of right now since it's documented. And also, the sub methods would be very small and I don't think we would benefit much from it. 

Let me know what you think 😃